### PR TITLE
Minor model operator simplifications

### PR DESCRIPTION
**Pull request by carmocca**
The only performance difference would be in the flash-attn case, which removed one redundant transpose.

Replacing `repeat_interleave` is just for the simpler operator.

### CODE REVIEW

File: lit_gpt/adapter.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ def forward(
             ay = self.scaled_dot_product_attention(q, ak, av, amask)
             y = y + self.gating_factor * ay
```
Comment (Andrei-Aksionov) [inline lit_gpt/adapter.py:254]: I am just eyeballing the code (sorry for not running on a pc prior to commenting), but this looks a bit off.

`ay` should have a shape something similar to `(B, nh, T, hs)`. Like it's describe in annotations in [lit-llama](https://github.com/Lightning-AI/lit-llama/blob/main/lit_llama/adapter.py#L164).
And gating factor now is `(1, 1, nh, 1)`. 
It looks like it should fail, shouldn't it?

Comment (Andrei-Aksionov) [reply lit_gpt/adapter.py:254]: P.S. In not so distant future I'll add annotation from `lit-llama` with some necessary updates.

Comment (carmocca) [reply lit_gpt/adapter.py:254]: The transpose is being done directly in `scaled_dot_product_attention` now

Comment (Andrei-Aksionov) [reply lit_gpt/adapter.py:254]: 😊 
Feel so embarrassed. I see `scaled_dot_product_attention` and immidiately think that it's from `torch.nn.functional` while it's a class method with the same name.
All I had to do is to simply scroll down ... :)

File: lit_gpt/adapter.py [hunk 2]
```diff
@@ -204,12 +204,12 @@ def forward(
...
-            k = k.repeat_interleave(q_per_kv, dim=2)
-            v = v.repeat_interleave(q_per_kv, dim=2)
+            k = k.expand(B, self.config.n_query_groups, q_per_kv, T, self.config.head_size)
+            v = v.expand(B, self.config.n_query_groups, q_per_kv, T, self.config.head_size)
 
         q = q.reshape(B, -1, T, self.config.head_size)  # (B, nh_q, T, hs)
-        k = k.view(B, -1, T, self.config.head_size)  # (B, nh_k, T, hs)
-        v = v.view(B, -1, T, self.config.head_size)  # (B, nh_v, T, hs)
+        k = k.reshape(B, -1, T, self.config.head_size)  # (B, nh_k, T, hs)
+        v = v.reshape(B, -1, T, self.config.head_size)  # (B, nh_v, T, hs)
```
Comment (awaelchli) [inline lit_gpt/adapter.py:212]: Why reshape over view here? Is this really necessary?

File: lit_gpt/adapter.py [hunk 3]
```diff
@@ -266,6 +266,16 @@ def forward(
     def reset_parameters(self) -> None:
         torch.nn.init.zeros_(self.gating_factor)
 
+    def _load_from_state_dict(self, state_dict: Dict, prefix: str, *args: Any, **kwargs: Any) -> None:
+        """For compatibility with older checkpoints."""
+        if (
```
[the remark this answers is not in the document]
Comment (carmocca) [reply lit_gpt/adapter.py:271]: True, but the tensor is the same whether you permute it or not in that instance

File: lit_gpt/adapter.py (hunk 2 above)
Comment (carmocca) [reply lit_gpt/adapter.py:212]: Because `expand` doesn't return a contiguous tensor, whereas `permute_interleave` did

**Pull request merged**